### ISSUE TITLE
When low poly models are used, Bambu Studio/X1 seems to overcompensate causing unusual artefacts.

### ISSUE THREAD

**Issue by RobinWhitfield** [bambulab/BambuStudio#228]
When a model is imported which is low poly so circles are not circles (and G2/G3 arcs are not being used), something seems to overcompensate on the shallow angle corners, causing an unusual and distinctive (and very obvious) surface finish.

I'm unsure whether this is an issue present on all shallow angle corners, or just when they're adjacent to a very short line.

Slowing down the print mitigates the issue to some extent.

This is not a general overextrusion issue as flow can be adjusted so that 90 degree corners are slightly underextruded and this issue is still present on curves.

Using a high poly model or a STEP file resolves the issue with curves.

![image](https://user-images.githubusercontent.com/63362209/189108109-f1f30afb-7699-491f-87d6-d241613d226c.png)

![image](https://user-images.githubusercontent.com/63362209/189108236-f32b63f0-2271-424d-841b-64ae28db95e5.png)
![image](https://user-images.githubusercontent.com/63362209/189108272-291095cb-6d1b-477d-8bc7-434612729dc3.png)

**Comment by 3dprintandpaint** [comment]
![IMG_20220908_051154274_HDR](https://user-images.githubusercontent.com/113126939/189178029-db325ec7-2c5b-4638-b346-503b5d11e78e.jpg)
Same problem here.  Left image is the Bambu X1C print, right is from a Prusa MK3S+.  High poly models are unaffected.

**Comment by 3dprintandpaint** [comment]
![Untitled](https://user-images.githubusercontent.com/113126939/189179568-ffb32479-b1ed-4b59-a2b5-7c526eaecfd0.jpg)

**Comment by 3dprintandpaint** [comment]
![Untitled](https://user-images.githubusercontent.com/113126939/189179707-d1181117-bfcb-4618-be97-ab00c4a0de07.jpg)

**Comment by YifanWuBambu** [comment]
@RobinWhitfield @3dprintandpaint 
Could you share us the models? Thanks!

**Comment by 3dprintandpaint** [comment]
> @RobinWhitfield @3dprintandpaint Could you share us the models? Thanks!

Hi, this is a paid model, however I have sought permission from the owner and they have kindly agreed for the file to be shared.  Hopefully this helps.
[31_base_ring.zip](https://github.com/bambulab/BambuStudio/files/9545958/31_base_ring.zip)

**Comment by YifanWuBambu** [comment]
@3dprintandpaint 
Thanks for your support. We will check this issue.

**Comment by RobinWhitfield** [comment]
My specific model was: https://www.thingiverse.com/thing:1930493

**Comment by psiberfunk** [comment]
This looks like an arc tolerance detection issue almost ?  I'm curious what the slice looks like in this case.  Do you guys have project files to share so we can make sure we're looking at exactly the same slice?

**Comment by RobinWhitfield** [comment]
These facets are quite large so I wouldn't expect the printer to try and turn them into an arc.

The slice looked as expected - lots of straight lines making up the curves

**Comment by psiberfunk** [comment]
> These facets are quite large so I wouldn't expect the printer to try and turn them into an arc.
> 
> The slice looked as expected - lots of straight lines making up the curves

So that's interesting, because when I used the default settings on your file , I got large arcs (although not the complete circle) like this:  
![image](https://user-images.githubusercontent.com/19777571/190513655-1652acb1-93f0-426d-9700-d7311c0c8098.png)

That's why i'm curious to see your project file.

**Comment by RobinWhitfield** [comment]
Oh, interesting. Let me try again with a new version of Bambu Studio and see what happens

**Comment by RobinWhitfield** [comment]
Nope, I still get lines on 01.02.00.09

![image](https://user-images.githubusercontent.com/63362209/190639895-b3e1f95a-b0d7-456d-9441-54a47daae0fd.png)

![image](https://user-images.githubusercontent.com/63362209/190639941-80baeb1e-6c63-4bd0-a9f6-fdfe7d6d6a0a.png)

**Comment by psiberfunk** [comment]
> The main issue however is how the printer seems to be responding to a series of short lines, rather than whether or not arcs are being generated.

So, forgive my confusion here ..is the gripe that even when the gcode is little lines it the print should be smoother? … or is the gripe that the slicer should be better at generating arcs in these situations?  

or is the gripe that even when the code is generating arcs that the firmware is parsing that into small line segments that show up ? (I don’t think this last one is the gripe but I included this possibility for completeness )

**Comment by RobinWhitfield** [comment]
The issue is that when the model is not being turned into arcs, the small line segments print poorly as there seems to be some kind of flow overcompensation. This is not an issue with most printers, where arcs aren't supported.

**Comment by psiberfunk** [comment]
Got it, thanks for clarifying!

**Comment by YifanWuBambu** [comment]
We are optimizing the vibration suppression algorithm.
For now, please slow down the outer wall speed to walkaround this issue.

**Comment by H4kunaM4tata** [comment]
@RobinWhitfield  I am afraid this is going to get nowhere.
My ticket isn't as bad, https://github.com/bambulab/BambuStudio/issues/1006 , but here are examples I have had to deal with:

If I tell somebody this was printed by an AUD$1.8k (~USD1.2k) printer, they will call me an idiot for spending that much money for such quality

At this moment I am avoiding models that have low poly and upscalling SLA models which has higher quality to workaround this problem.
MK3S+ wasn't perfect but I never ever had such a terrible quality print like this

![20221118_002239](https://user-images.githubusercontent.com/35049512/211177297-631039c3-dd49-4c3a-a819-62def5e6b561.jpg)

![20221118_001841](https://user-images.githubusercontent.com/35049512/211177304-f0971b36-c93b-4023-bbfc-e9c800f6a1a2.jpg)

This problem is way beyond BambuLab wants to recognize since it also affect flat walls.
It took me 3 months, that is correct, 3 months and countless tests, countless filament wasted, countless f words thrown at the printer to get a print worth $1.8k

![image](https://user-images.githubusercontent.com/35049512/211177535-f6de4a06-b42f-47ee-80ea-1d13149e3c48.png)

![image](https://user-images.githubusercontent.com/35049512/211177539-11c1d10e-4185-43ef-bce0-a7b7b00fc35e.png)

And yet, same model, same everything, few days apart.
On this case, the right model quality was affected because the CF rod requires almost everyday cleaning with isopropryl alcohol or it starts to create a wild resonance which is then transferred to the print.

![PXL_20230108_020958754](https://user-images.githubusercontent.com/35049512/211177583-fd419132-efe2-4d38-af5a-ecff1a752354.jpg)

It is safe to say I don't have a top quality printer but an Ender printer that requires me to constantly babysit it to print right.
